README.md

```diff
@@ -1,3 +1,166 @@
+    priority: <priority>                # profile ordering priority, lower numbers mean higher priority (0 is the highest priority)
+    profile: <tuned_profile_name>       # e.g. tuned_profile_1
+
+  # ...
+
+  - match:
+    <match>
+    priority: <priority>
+    profile: <tuned_profile_name>       # e.g. tuned_profile_n
+```
+
+If `<match>` is omitted, a profile match (i.e. _true_) is assumed.
+
+`<match>` is an optional array recursively defined as follows:
+
+```
+    - label: <label_name>     # node or pod label name
+      value: <label_value>    # optional node or pod label value; if omitted, the presence of <label_name> is enough to match
+      type: <label_type>      # optional node or pod type ("node" or "pod"); if omitted, "node" is assumed
+      <match>                 # an optional <match> array
+```
+
+If `<match>` is not omitted all nested `<match>` sections must also
+evaluate to _true_, otherwise _false_ is assumed and the profile with
+the respective `<match>` section will not be applied/recommended.
+Therefore the nesting (child `<match>` sections) works as logical
+AND operator.  Conversely, if any item of the `<match>` array matches,
+the entire `<match>` array evaluates to _true_, therefore the array
+acts as logical OR operator.
+
+#### Example
+
+```
+  - match:
+    - label: tuned.openshift.io/elasticsearch
+      match:
+      - label: node-role.kubernetes.io/master
+      - label: node-role.kubernetes.io/infra
+      type: pod
+    priority: 10
+    profile: openshift-control-plane-es
+  - match:
+    - label: node-role.kubernetes.io/master
+    - label: node-role.kubernetes.io/infra
+    priority: 20
+    profile: openshift-control-plane
+  - priority: 30
+    profile: openshift-node
+```
+
+The CR above is translated for the containerized tuned daemon into
+its recommend.conf file based on the profile priorities.  The profile
+with the highest priority (10) is openshift-control-plane-es and
+therefore it is considered first.  The containerized tuned daemon
+running on a given node looks to see if there is a pod running on the
+same node with tuned.openshift.io/elasticsearch label set.  If not,
+the entire `<match>` section evaluates as _false_.  If there is such a
+pod with the label, in order for the `<match>` section to evaluate to
+_true_ the node label also needs to be node-role.kubernetes.io/master
+OR node-role.kubernetes.io/infra.
+
+If the labels for the profile with priority 10 matched,
+openshift-control-plane-es profile is applied and no other profile is 
+considered.  In case the node/pod label combination did not match, 
+the second highest priority profile (openshift-control-plane) is considered.
+This profile is applied if the containerized tuned pod runs on a node with 
+labels node-role.kubernetes.io/master OR node-role.kubernetes.io/infra.
+
+Finally, the profile `openshift-node` has the lowest priority of 30.
+It lacks the `<match>` section and therefore will always match and
```

Comment:
, therefore,

Reply:
match. It acts as a profile catch-all to set the  `openshift-node` profile, if no

Reply:
What do you mean by a catch-all?

Reply:
To me that means something that ends up being used if nothing else "matches" in this case.  Last resort.  For example, if none of the node/pod labels allow for a higher priority profile to be set, the lowest priority (catch-all) profile is used.

Reply:
@jmencak Okay, that makes more sense. Thanks!